### DISCUSSION TITLE
Unclear how to set password

### DISCUSSION THREAD

**Discussion by Ren656** [Koenkk/zigbee2mqtt#25338] [General]
as a noob i followed the instructions of HomeAutomation Guy to install HA in Docker on raspberry 4b, including MQTT and Zigbee2MQTT. I was able to set a password for MQTT in a MQTT Portainer terminal at prompt /# typing ‘mosquitto_passwd -c /mosquitto/config/password.txt hass’. That one was working. But setting a password for Zigbee2MQTT: i can’t figure out how to following the instructions according to https://www.zigbee2mqtt.io/guide/configuration/mqtt.html#specifying-mqtt-server-user-password-and-network-key-in-a-different-file. Either way, most logical to me is to get user name and password from the existing file password.txt but how?

**Comment by MSL-DA** [comment]
Look here: https://www.zigbee2mqtt.io/guide/configuration/frontend.html

`To specify the auth_token in a different file set e.g. auth_token: '!secret.yaml auth_token', create a file called secret.yaml next to configuration.yaml with content auth_token: super-secret-token`

**Comment by Ren656** [comment]
Dag Koen,

Erg bedankt voor je snelle antwoord. Vraagje: welke configuration.yaml moet ik hebben, die van Homeassistant of die van Zigbee2MQTT. Tot nu toe ging ik er van uit dat ik die van Zigbee2MQTT moest hebben. Password.txt zit overigens in de map van mosquitto.

Vriendelijke groet,
René
Verstuurd vanaf mijn iPad

Op 27 dec 2024 om 18:01 heeft MSL-DA ***@***.***> het volgende geschreven:

﻿

Look here: https://www.zigbee2mqtt.io/guide/configuration/frontend.html

To specify the auth_token in a different file set e.g. auth_token: '!secret.yaml auth_token', create a file called secret.yaml next to configuration.yaml with content auth_token: super-secret-token

**Reply by MSL-DA** [reply]
I don't use z2m addon in HA. Otherwise it is in your Zigbee2MQTT data folder > configuration.yaml

You can try to open z2m frontend>settings>frontend and add a code under "Auth token" this will be stored in Zigbee2MQTT > configuration.yaml
